src/components/sidecar/use-active-section.ts

```diff
@@ -42,22 +67,32 @@ export function useActiveSection(
           ? findMatchingSectionIndex(entries[0].target.id)
           : -1
 
-        // Activate only the bottom-most visible section heading
-        if (visibleHeadings.length === 1) {
-          setActiveSection(visibleHeadings[0])
-        } else if (visibleHeadings.length > 1) {
-          setActiveSection(visibleHeadings[visibleHeadings.length - 1])
-        }
-
-        if (previousY.current) {
-          // If we detect that we're scrolling up, and there are no visible headers,
-          // optimistically set the previous header as visible to make the active section match the visible content
-          if (visibleHeadings.length === 0 && scrollTrend === 'up') {
+        // Find the heading closest to the top
+        const visibleHeadingIds = Object.keys(visibleHeadings.current)
+        if (visibleHeadingIds.length > 0) {
+          let shortestDistance
+          let closestHeading
+          visibleHeadingIds.forEach((headingId) => {
+            const headingEntry = visibleHeadings.current[headingId]
+            const distance = headingEntry.getBoundingClientRect().bottom
+            if (!closestHeading || distance < shortestDistance) {
+              closestHeading = headingId
+              shortestDistance = distance
+            }
+          })
+          setActiveSection(closestHeading)
```

Comment:
Thinking out loud...I wonder this could be accomplished by tweaking the root margin instead? If we increase the margin to something like `-80`, there would in theory be less collisions where a heading lower down the page would intersect. The root margin viz in this article was helpful for my mental model: https://blog.arnellebalane.com/the-intersection-observer-api-d441be0b088d

Reply:
I agree with this, I think we can actually start refining this value now that we've got a better hold on tracking visible headings! I'll up that bottom rootMargin value and we can test it out. 😊

Reply:
Gave some numbers a try, and `-60%` on the bottom seems like the safest bet. `-80%` was too large with the top value now being set, and `-65%` seemed like the smallest value that would work with the very last headings on a page. I think `-60%` gives us a little more room for those bottom headings.